### PR TITLE
Rdiff backup tests

### PR DESCRIPTION
**Pull request by hsolorza**
This includes most of the tests for rdiff_backup. There are a few tests in server_spec that are commented out as they do not work yet. Also, if you run `rake unit` while on this branch, there are a few more resources that still need to be tested. I have yet to find a way to properly test these 5 resources so let me know if you have any tips.

### CODE REVIEW

**Comment by hsolorza** [comment]
@osuosl-cookbooks/chefs

File: spec/rdiff-backup-test/server_spec.rb [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        # it do
+        #   expect(chef_run).to create_file('check_rdiff_job_test1').with(
+        #   command: '/usr/bin/sudo/' + chef_run.node['nrpe']['plugin_dir'] +
+        #            'check_rdiff' + "-w 16 "\
+        #                            "-c 18 "\
+        #                            "-r /data/rdiff-backup "\
+        #                            "-p 24 "\
+        #                            "-l 800000000"
+        #   )
+        # end
```
Comment (ramereth) [inline spec/rdiff-backup-test/server_spec.rb]: I believe you need to step into the nrpe_check resource to make this test work.

File: spec/rdiff-backup-test/server_spec.rb [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+              group: 'rdiff-backup-server' || 'rdiff-backup-server', # Group || Owner
+              mode: 0700
+            )
+          end
+          # it do
+          #   expect(chef_run).to create_ssh_user('id_rsa').with(
+          #   user: 'rdiff-backup-server'
+          #   key: secrets['ssh-key']
+          #   )
+          # end
```
Comment (ramereth) [inline spec/rdiff-backup-test/server_spec.rb]: I believe you need to step into the ssh_user_private_key resource to make this work.

File: spec/rdiff-backup-test/server_spec.rb [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+      #   expect(chef_run).to remove_nrpe_check('check_rdiff_job_test1')
+      # end
+      # context 'Delete: nrpe is false' do
+      #   let(:runner) do
+      #     ChefSpec::SoloRunner.new(
+      #       pltfrm.dup.merge(step_into: ['nrpe-check'])
+      #     )
+      #   end
+      #   before do
+      #     chef_run.node.default['rdiff-backup']['nrpe'] = false
```
Comment (hsolorza) [inline spec/rdiff-backup-test/server_spec.rb]: @ramereth Im still having difficulty doing the nrpe tests. I looked in the upstream cookbook to see if they have any matchers but they did not write any tests for the nrpe_check, all I could find on the checks was these. `https://github.com/schubergphilis/nrpe/blob/master/test/fixtures/cookbooks/chef_nrpe_test/recipes/default.rb`  `https://github.com/schubergphilis/nrpe/blob/master/resources/check.rb`  Could I make a matcher for this?

**Comment by ramereth** [comment]
[rdiff-backup](https://trello.com/c/0ijYmIaB/147-rdiff-backup)

**Comment by lo9key** [comment]
[rdiff-backup](https://trello.com/c/0ijYmIaB/147-rdiff-backup)

**Comment by lo9key** [comment]
Closing this PR because this work was finished in [another PR](https://github.com/osuosl-cookbooks/rdiff-backup/pull/44).

**Pull request closed without merging**